test/testsymboldatabase.cpp

```diff
@@ -1264,6 +1265,22 @@ class TestSymbolDatabase : public TestFixture {
         ASSERT(p->valueType()->pointer == 1);
     }
 
+    void VariableValueType6() {
+        GET_SYMBOL_DB("struct Data{};\n"
+                      "void foo(std::shared_ptr<C>* p) { std::unique_ptr<Data> data = std::unique_ptr<Data>(new Data); }");
+
+        const Token * check{nullptr};
```

Comment:
you can replace the for loop:

        const Token* check = Token::findsimplematch(tokenizer.tokens(), "( new");
        ASSERT(check);
        ASSERT(check->valueType());
        ASSERT(check->valueType()->smartPointerTypeToken);